examples/prodlda.py

```diff
@@ -0,0 +1,207 @@
+from sklearn.feature_extraction.text import CountVectorizer
+from wordcloud import WordCloud
+
+import haiku as hk
+import jax
+from jax import device_put, random
+import jax.numpy as jnp
+
+import numpyro
+from numpyro.contrib.module import haiku_module
+import numpyro.distributions as dist
+from numpyro.infer import SVI, TraceMeanField_ELBO
+
+
+class HaikuEncoder:
+    def __init__(self, vocab_size, num_topics, hidden, dropout_rate):
+        self._dropout_rate = dropout_rate
+        self._vocab_size = vocab_size
+        self._hidden = hidden
+        self._num_topics = num_topics
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+
+        h = jax.nn.softplus(hk.Linear(self._hidden)(inputs))
+        h = jax.nn.softplus(hk.Linear(self._hidden)(h))
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, h)
+
+        # NB: here we set `create_scale=False` and `create_offset=False` to reduce
+        # the number of learning parameters
+        h_mu = hk.Linear(self._num_topics)(h)
+        logtheta_loc = hk.BatchNorm(
+            create_scale=False, create_offset=False, decay_rate=0.1
+        )(h_mu, is_training)
+
+        h_sigma = hk.Linear(self._num_topics)(h)
+        logtheta_logvar = hk.BatchNorm(
+            create_scale=False, create_offset=False, decay_rate=0.1
+        )(h_sigma, is_training)
+        logtheta_scale = jnp.exp(0.5 * logtheta_logvar)
+        return logtheta_loc, logtheta_scale
+
+
+class HaikuDecoder:
+    def __init__(self, vocab_size, dropout_rate):
+        self._dropout_rate = dropout_rate
+        self._vocab_size = vocab_size
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, inputs)
+        h = hk.Linear(self._vocab_size)(h)
+        h = hk.BatchNorm(create_scale=True, create_offset=True, decay_rate=0.1)(
+            h, is_training
+        )
+        return jax.nn.softmax(h)
+
+
+def model(docs, hyperparams, is_training=False):
+    decoder = haiku_module(
+        "decoder",
+        # use `transform_with_state` for BatchNorm
+        hk.transform_with_state(
+            HaikuDecoder(hyperparams["vocab_size"], hyperparams["dropout_rate"])
+        ),
+        input_shape=(1, hyperparams["num_topics"]),
+        apply_rng=True,
+        # to ensure proper initialisation of BatchNorm we must
+        # initialise with is_training=True
+        is_training=True,
+    )
+
+    with numpyro.plate(
+        "documents", docs.shape[0], subsample_size=hyperparams["batch_size"]
+    ):
+        batch_docs = numpyro.subsample(docs, event_dim=1)
+        logtheta_loc = jnp.zeros((batch_docs.shape[0], hyperparams["num_topics"]))
+        logtheta_scale = jnp.ones((batch_docs.shape[0], hyperparams["num_topics"]))
+        logtheta = numpyro.sample(
+            "logtheta", dist.Normal(logtheta_loc, logtheta_scale).to_event(1)
```

Comment:
You can use `dist.Normal(0, 1).expand([hyperparams["num_topics"]]).to_event(1)` for simplicity.

Reply:
Very slick 🙂